source/dofs/dof_tools.cc

```diff
@@ -1133,6 +1133,43 @@ namespace DoFTools
 
 
 
+  template <int dim, int spacedim>
+  void
+  extract_locally_active_level_dofs(
+    const DoFHandler<dim, spacedim> &dof_handler,
+    IndexSet &                       dof_set,
+    const unsigned int               level)
+  {
+    // collect all the locally owned dofs
+    dof_set = dof_handler.locally_owned_mg_dofs(level);
+
+    // add the DoF on the adjacent ghost cells to the IndexSet, cache them
+    // in a set. need to check each dof manually because we can't be sure
+    // that the dof range of locally_owned_dofs is really contiguous.
+    std::vector<types::global_dof_index> dof_indices;
+    std::set<types::global_dof_index>    global_dof_indices;
+
+    typename DoFHandler<dim, spacedim>::level_cell_iterator
+      cell = dof_handler.begin_mg(level),
+      endc = dof_handler.end_mg(level);
+    for (; cell != endc; ++cell)
```

Comment:
Would you mind converting this to a range-based for loop?
```suggestion
    for (const auto& cell: dof_handler.mg_cell_iterators_on_level(level))
```

Reply:
@masterleinad Thanks for the feedback! Especially the lambda solution is much nicer than the pointer one.
Regarding the range-based for loop I wasn't sure how backwards compatible one should stay but changed it now to a filtered iterator and range based for loop solution.